### PR TITLE
Section retrieval API

### PR DESCRIPTION
**Pull request by gwicke**
Task: https://phabricator.wikimedia.org/T94890

### CODE REVIEW

File: mods/parsoid.js [hunk 1]
```diff
@@ -69,17 +70,67 @@ PSP.wrapContentReq = function(restbase, req, promise) {
...
         // if we have reached this point,
         // it means access is not denied
-        return ensureCharsetInContentType(resx[0]);
+        if (format === 'html' && req.query.sections && responses.sectionOffsets) {
+            // Handle section requests
+            var sectionOffsets = responses.sectionOffsets.body;
+            var sections = req.query.sections.split(',').map(function(id) {
+                return id.trim();
+            });
+            var body = cheapBodyInnerHTML(responses.content.body.toString());
```
Comment (d00rman) [inline mods/parsoid.js]: Is it a fact that section offsets provided by Parsoid will always match our cheap body return? Probably worth to put a comment about it, if so.

File: mods/parsoid.js [hunk 2]
```diff
@@ -69,17 +70,67 @@ PSP.wrapContentReq = function(restbase, req, promise) {
...
+
+            return {
+                status: 200,
+                headers: {
+                    etag: responses.content.headers.etag,
+                    'content-type': 'application/json',
+                },
+                body: chunks,
+            };
+        } else {
```
Comment (d00rman) [inline mods/parsoid.js]: I think an `else if` covering the case where `req.query.sections` has been specified, but no section offsets have been returned is necessary here seeing that any errors from that call are ignored (as stated on line 94 above).

**Comment by d00rman** [comment]
A couple of minor comments, otherwise LGTM

File: mods/parsoid.js (hunk 1 above)
Comment (gwicke) [inline mods/parsoid.js]: They are relative to body.innerHTML.

File: mods/parsoid.js (hunk 2 above)
Comment (gwicke) [inline mods/parsoid.js]: I handled this differently by no longer catching the section offset request error. That way it's propagated the same way revision metadata 403s are.

**Pull request merged**